packages/utils/src/assistant/index.ts

```diff
@@ -54,10 +54,12 @@ const assign = (...sources: AssistantDefinition[]): AssistantDefinition => {
         ...(typeof acc.config.defaultSeverity === 'undefined'
           ? {}
           : { defaultSeverity: acc.config.defaultSeverity }),
-        rules: {
-          ...curr.config.rules,
-          ...acc.config.rules,
-        },
+        rules: Object.entries(curr.config.rules).reduceRight(
+          (a: { [key: string]: any }, [name, opts]) => {
```

Comment:
Is it feasible to avoid `any` here? Could maybe use `Record`?

```
Record<string, RuleConfigOption>
```

or similar?